python/cudf/cudf/core/column/column.py

```diff
@@ -2318,285 +2333,69 @@ def as_column(
         )
     elif isinstance(arbitrary, cudf.Scalar):
         data = ColumnBase.from_scalar(arbitrary, length if length else 1)
+    elif can_memoryview(arbitrary):
+        data = as_column(
+            memoryview(arbitrary), dtype=dtype, nan_as_null=nan_as_null
+        )
```

Comment:
> I think I would prefer if all of these branches return ... rather than assigning to a variable and going to fallthrough.

Good point. I'll try to have every branch do `return ...` for clarity. One aspect I'm not sure if when `dtype=` is specified that an "astype" is done if the input is typed for all branches